v1.0/contribute-to-cockroachdb.md

```diff
@@ -4,20 +4,95 @@ summary: Contribute to the development of CockroachDB.
+
+### Low Complexity Projects
+
+These cover things like bug fixes and small enhancements that don’t involve any major architectural or design decisions. You should feel free to submit these for review as you come across them.
+
+1. Find or create an issue.
+2. Write the fix, include the appropriate tests, and follow our [code review workflow](https://github.com/cockroachdb/cockroach/blob/master/CONTRIBUTING.md#code-review-workflow).
+3. A CockroachDB engineer will validate and merge.
+
+A good place to find these types of projects are open issues with the label ["easy"](https://github.com/cockroachdb/cockroach/issues?q=is%3Aopen+is%3Aissue+label%3Aeasy+label%3Ahelpwanted).
+
+### Medium Complexity Projects
+
+For these projects, you should:
+
+1. Proactively reach out to the Cockroach team and discuss your proposal.
+2. One of our engineers will provide feedback on your design. Occasionally, you will be asked to contribute an RFC, too, in which case you should follow the regular [RFC process](https://github.com/cockroachdb/cockroach/tree/master/docs/RFCS) before you start implementing the solution.
+3. When everyone is happy, you can get started coding and follow the normal [code review workflow](https://github.com/cockroachdb/cockroach/blob/master/CONTRIBUTING.md#code-review-workflow).
+
+{{site.data.alerts.callout_info}}If we are under time pressure to deliver a feature and it’s not tracking towards completion (or if your team goes radio silent), we may pull development back in house.{{site.data.alerts.end}}
+
+Here’s a great example of a contributor [building support for outer joins](https://github.com/cockroachdb/cockroach/issues/13342) in partnership with our engineering team.
+
+### High Complexity Projects
+
+These projects involve fundamental changes to the way CockroachDB works. What’s important to remember here is that supporting external development of these projects will take a non-trivial amount of time from the Cockroach team that would otherwise be spent building the product, so we really need to make sure it’s worth it. In practice, we’ve found even strong developers have little success with these types of projects and, in most cases, it is faster for our team to build these features ourselves.
+
+That being said, if you’re up for the challenge there are a few prerequisites. You should:
+
+1. Convince the CockroachDB team of your ability to drive a larger project. For this, you can either work with the CockroachDB team on mid-sized projects first, or provide some other evidence that you are an expert at building high quality distributed systems.
+    {{site.data.alerts.callout_info}}The engineering team may not support further exploration if they think the project is too complex for an external contributor or if we already have a schedule for this work and you can't guarantee delivery on that schedule.{{site.data.alerts.end}}
```

Comment:
"or if we already have this work scheduled" vs "if we already have a schedule for this work"

Reply:
Done.